concrete-core/src/specification/engines/lwe_keyswitch_key_creation.rs

```diff
@@ -13,6 +13,28 @@ engine_error! {
                               the precision of the ciphertext."
 }
 
+impl<EngineError: std::error::Error> LweKeyswitchKeyCreationError<EngineError> {
+    pub fn perform_generic_checks(
+        decomposition_level_count: DecompositionLevelCount,
+        decomposition_base_log: DecompositionBaseLog,
+        integer_precision: usize,
+    ) -> Result<(), Self> {
+        if decomposition_base_log.0 == 0 {
+            return Err(Self::NullDecompositionBaseLog);
+        }
+
+        if decomposition_level_count.0 == 0 {
```

Comment:
Earlier the condition was l == 1, why is it different here? Was the == 1 a mistake?

Reply:
I'd say the `==1` is an error and it should be `==0` as `null` stands for `0`, but we need to double-check that

Reply:
Yes, reading from the error description, we should check for zero. Don't know why it ended up this way. Maybe at some point I thought it would not make much sense to have a decomposition of size one ?